### DISCUSSION TITLE
Replacing YAML Value in a different note

### DISCUSSION THREAD

**Discussion by gruff101** [SilentVoid13/Templater#1115] [Help]
Looked at the thread for replacing values in a different note. The value is in YAML.

https://github.com/SilentVoid13/Templater/discussions/1107

Unfortunately find and replace will not work as I don't always know the value of the YAML variable I am replacing. I am able to pull the value back stored in YAML however anything I update is a copy and therefore doesn't update the note itself.

Does anyone know how to update a YAML value in a different note to the current one?

**Comment by Zachatoo** [comment]
Obsidian has a built in `app.fileManager.processFrontMatter()` method for precisely this purpose.

```js
<%*
const file = tp.file.find_tfile("Note you want to update frontmatter for");
await app.fileManager.processFrontMatter(file, (frontmatter) => {
  // Modify frontmatter of another note here
  frontmatter["status"] = "In progress";
  frontmatter["review count"] += 1;
  delete frontmatter["ignored"];
});
-%>
```

Relevant links:

https://zachyoung.dev/snippets/templater#update-frontmatter
